suricata/update/parsers.py

```diff
@@ -56,13 +61,17 @@
       'help': "Override Suricata version"}),
     (("--user-agent",),
      {'metavar': '<user-agent>',
-      'help': "Set custom user-agent string"}),
+      'help': "Set custom user-agent string" 
+      if show_advanced else argparse.SUPPRESS}),
     (("--no-check-certificate",),
      {'action': 'store_true', 'default': None,
       'help': "Disable server SSL/TLS certificate verification"}),
     (("-V", "--version"),
      {'action': 'store_true', 'default': False,
-      'help': "Display version"})
+      'help': "Display version"}),
+    (("-s","--show-all-help"),
```

Comment:
I think I'd rather have this as "--show-advanced" with a help of "Show advanced options".